server/src/models/productModel.ts

```diff
@@ -0,0 +1,27 @@
+import { Schema, model, Types } from "mongoose";
+
+interface Exercise {
+  _id: Types.ObjectId;
+  name: string;
+  sets?: number;
+  reps?: number;
+}
+
+interface ExerciseSet {
+  _id: Types.ObjectId;
+  date: Date;
+  exercises: Types.Array<Exercise>;
+}
+
+const setSchema = new Schema<Exercise>({
```

Comment:
Perhaps `const ExerciseSchema = new Schema`? This way it describes what is this schema about. `setSchema` suggests this is a method

Reply:
Changed to DaySchema and ExerciseSchema